packages/component/src/Icon/ComponentIcon.tsx

```diff
@@ -0,0 +1,35 @@
+import { validateProps } from 'botframework-webchat-react-valibot';
+import { useStyles } from 'botframework-webchat-styles/react';
+import cx from 'classnames';
+import React from 'react';
+import { object, optional, pipe, readonly, string, type InferInput } from 'valibot';
+
+import createIconComponent from '../Utils/createIconComponent';
+import styles from './ComponentIcon.module.css';
+
+const componentIconPropsSchema = pipe(
+  object({
+    className: optional(string()),
+    icon: optional(string()),
+    appearance: optional(string()),
+    size: optional(string())
+  }),
+  readonly()
+);
+
+type ComponentIconProps = InferInput<typeof componentIconPropsSchema>;
+
+function BaseComponentIcon(props: ComponentIconProps) {
+  const { className } = validateProps(componentIconPropsSchema, props);
+
+  const classNames = useStyles(styles);
+
+  return <div className={cx(classNames['component-icon'], className)} />;
+}
+
+const ComponentIcon = createIconComponent(styles, BaseComponentIcon);
+
+ComponentIcon.displayName = 'ComponentIcon';
+
+export default ComponentIcon;
+export type { ComponentIconProps };
```

Comment:
I am actually if we should expose the props as schema, but we are doing it today, so.

```suggestion
export { componentIconPropsSchema, type ComponentIconProps };
```